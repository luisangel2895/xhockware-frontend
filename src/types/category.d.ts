export interface Category {
  name: string;
  status: boolean;
}

export interface AllCategories {
  all: {
    name: string;
    status: boolean;
  };
  health: {
    name: string;
    status: boolean;
  };
  economy: {
    name: string;
    status: boolean;
  };
  tecnology: {
    name: string;
    status: boolean;
  };
  politics: {
    name: string;
    status: boolean;
  };
  ecology: {
    name: string;
    status: boolean;
  };
  social: {
    name: string;
    status: boolean;
  };
}
