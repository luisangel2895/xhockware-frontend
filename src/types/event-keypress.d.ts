export interface EventKeyPress {
  isTrusted: boolean;
  altKey: boolean;
  bubbles: boolean;
  cancelBubble: boolean;
  cancelable: boolean;
  charCode: number;
  code: string;
}
