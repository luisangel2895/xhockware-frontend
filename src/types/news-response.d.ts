export interface NewsResponse {
  status: string;
  totalResults: number;
  articles: News[];
}

export interface News {
  source: Source;
  author: string;
  category: string;
  title: string;
  description: string;
  url: string;
  urlToImage: string;
  publishedAt: string;
  content: string;
}

export interface Source {
  id: string;
  name: string;
}
