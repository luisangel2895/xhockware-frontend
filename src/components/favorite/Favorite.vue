<template lang="pug">
.favorite
	.favorite__img
		.img-container(v-if="news")
			img(:src="news.urlToImage")
	.favorite__text
		.favorite__text-category {{filterCategory()}}
		.favorite__text-title {{filterLengthText()}}
</template>
<script lang="ts">
import { Options, Vue } from "vue-class-component";
// Types
import { News } from "@/types/news-response";

@Options({
  props: {
    news: Object,
  },
})
export default class Favorite extends Vue {
  news!: News;

  filterLengthText(): string {
    if (this.news) {
      if (this.news.title.length <= 43) {
        return this.news.title;
      } else {
        return this.news.title.slice(0, 40) + " ...";
      }
    } else {
      return "";
    }
  }
  filterCategory(): string {
    if (this.news) {
      const word = this.news.category;
      return word.charAt(0).toUpperCase() + word.slice(1);
    } else {
      return "";
    }
  }
}
</script>
<style lang="scss" scoped>
.favorite {
  width: 100%;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  border-radius: 20px;
  &__img {
    width: 30%;
    img {
      width: 100%;
      height: 100px;
      object-fit: cover;
      border-radius: 20px;
    }
  }
  &__text {
    width: 70%;
    padding: 20px;
    &-category {
      font-family: $regular;
      font-size: 1.4rem;
      color: $grey_primary;
      margin-bottom: 10px;
    }
    &-title {
      font-family: $semibold;
      font-size: 1.6rem;
      color: $black_primary;
    }
  }
}
// Tablet
@media (min-width: 768px) {
  .favorite {
    width: 33.3%;
  }
}
// PC
@media (min-width: 1300px) {
  .favorite {
    width: 100%;
  }
}
</style>
