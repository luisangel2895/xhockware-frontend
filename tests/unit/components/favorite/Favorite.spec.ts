import { shallowMount } from "@vue/test-utils";
// Components
import Favorite from "@/components/favorite/Favorite.vue";
// Mocks
import { newsDetailMock } from "../../mocks/news.mocks";
// Types
import { News } from "@/types/news-response";

const wrapper = shallowMount(Favorite, {
  props: {
    news: newsDetailMock as News,
  },
});

describe("Test in view Favorite.vue", (): void => {
  it("Match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Elements (.favorite, .favorite__img, img, .favorite__text) are mounted correctly", (): void => {
    expect(wrapper.find(".favorite").exists()).toBeTruthy();
    expect(wrapper.find(".favorite__img").exists()).toBeTruthy();
    expect(wrapper.find("img").exists()).toBeTruthy();
    expect(wrapper.find(".favorite__text").exists()).toBeTruthy();
  });
  it("Data of news is correctly setted in this component", (): void => {
    // The image of the news
    expect(wrapper.find("img").attributes("src")).toBe(
      "https://images.wsj.net/im-466867/social"
    );
    // Title
    expect(wrapper.find(".favorite__text-title").text()).toBe(
      "Second Life Founder Returns to Take On t ..."
    );
    // Category
    expect(wrapper.find(".favorite__text-category").text()).toBe("Tecnology");
  });
});
