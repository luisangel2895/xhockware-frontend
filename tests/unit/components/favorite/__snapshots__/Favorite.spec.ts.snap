// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in view Favorite.vue Match with snapshot 1`] = `
<div class="favorite">
  <div class="favorite__img">
    <div class="img-container"><img src="https://images.wsj.net/im-466867/social"></div>
  </div>
  <div class="favorite__text">
    <div class="favorite__text-category">Tecnology</div>
    <div class="favorite__text-title">Second Life Founder Returns to Take On t ...</div>
  </div>
</div>
`;
